Makefile

```diff
@@ -0,0 +1,103 @@
+APP_NAME = mobile-security-service-server
+ORG_NAME = aerogear
+PKG = github.com/$(ORG_NAME)/$(APP_NAME)
+APP_FILE=./cmd/mobile-security-service-server/main.go
+TOP_SRC_DIRS = pkg
+PACKAGES     ?= $(shell sh -c "find $(TOP_SRC_DIRS) -name \\*_test.go \
+                   -exec dirname {} \\; | sort | uniq")
+BIN_DIR := $(GOPATH)/bin				   
+BINARY ?= mobile-security-service-server
+
+# This follows the output format for goreleaser
+BINARY_LINUX_64 = ./dist/linux_amd64/$(BINARY)
+
+DOCKER_LATEST_TAG = $(ORG_NAME)/$(APP_NAME):latest
+DOCKER_MASTER_TAG = $(ORG_NAME)/$(APP_NAME):master
+RELEASE_TAG ?= $(CIRLE_TAG)
+DOCKER_RELEASE_TAG = $(ORG_NAME)/$(APP_NAME):$(RELEASE_TAG)
+
+LDFLAGS=-ldflags "-w -s -X main.Version=${TAG}"
+
+.PHONY: setup
+setup:
+	dep ensure
+
+.PHONY: build
+build: setup
+	go build -o $(BINARY) $(APP_FILE)
+
+.PHONY: build_linux
+build_linux: setup
+	env GOOS=linux GOARCH=amd64 go build -o $(BINARY_LINUX_64) $(APP_FILE)
+
+.PHONY: docker_build
+docker_build: build_linux
+	docker build -t $(DOCKER_LATEST_TAG) --build-arg BINARY=$(BINARY_LINUX_64) .
+
+.PHONY: docker_build_release
+docker_build_release: build_linux
+	docker build -t $(DOCKER_RELEASE_TAG) --build-arg BINARY=$(BINARY_LINUX_64) .
```

Comment:
fyi: 
```
╭─ ~/go-code/src/github.com/aerogear/mobile-security-service-server ‹AEROGEAR-8397› 
╰─$ make docker_build_release 
dep ensure
Warning: the following project(s) have [[constraint]] stanzas in Gopkg.toml:

  ✗  github.com/jmoiron/sqlx
  ✗  github.com/lib/pq

However, these projects are not direct dependencies of the current project:
they are not imported in any .go files, nor are they in the 'required' list in
Gopkg.toml. Dep only applies [[constraint]] rules to direct dependencies, so
these rules will have no effect.

Either import/require packages from these projects so that they become direct
dependencies, or convert each [[constraint]] to an [[override]] to enforce rules
on these projects, if they happen to be transitive dependencies.

env GOOS=linux GOARCH=amd64 go build -o ./dist/linux_amd64/mobile-security-service-server ./cmd/mobile-security-service-server/main.go
docker build -t aerogear/mobile-security-service-server: --build-arg BINARY=./dist/linux_amd64/mobile-security-service-server .
invalid argument "aerogear/mobile-security-service-server:" for "-t, --tag" flag: invalid reference format
See 'docker build --help'.
make: *** [Makefile:39: docker_build_release] Error 125

```

Reply:
@damienomurchu this is just because these packages are included but not yet used by the application. I will remove them though until they are used.